storage/StorageImpl.cc

```diff
@@ -445,4 +443,88 @@ namespace storage
 	// TODO somehow update probed
     }
 
+
+    void
+    Storage::Impl::generate_pools(const Devicegraph* devicegraph)
+    {
+	ST_CHECK_PTR(devicegraph);
+
+	// TODO more types, e.g. dasd, multipath, pmem, nvme?
+	// TODO check partition table?
+	// TODO do not add already existing devices
+
+	for (const Disk* disk : Disk::get_all(devicegraph))
+	{
+	    string name = disk->is_rotational() ? "HDDs" : "SSDs";
```

Comment:
Will any USB sticks / SD cards that happen to be plugged in also appear in the "SSDs" pool? Or in a separate one? Or in none?

Reply:
They will also appear as SSDs. What pools should be generated still has to be discussed.